### PR TITLE
logrus logging bridge

### LINKED ISSUE #5193: Add logrus log bridge
Add [`logrus.Hook`](https://pkg.go.dev/github.com/sirupsen/logrus#Hook) log bridge.

### PR DESCRIPTION
**Pull request by dmathieu**
Closes #5193.

This adds the logrus log bridge

### CODE REVIEW

**Review by hanyuancheung — APPROVED** [review]

File: bridges/otellogrus/hook.go
Comment (pellared) [inline bridges/otellogrus/hook.go:148]:
```suggestion
func NewHook(options ...Option) *Hook {
```

File: bridges/otellogrus/hook.go
Comment (pellared) [inline bridges/otellogrus/hook.go:188]: Just a remark that we could use a `kvBuffer` like in `otelslog` but I prefer to do it as it is currently and focus on performance improvements in separate PRs.

File: bridges/otellogrus/hook.go
Comment (pellared) [inline bridges/otellogrus/hook.go:197]: Why not `kvs := make([]log.KeyValue, 0, len(fields))` and simply appending?

```suggestion
	kvs := make([]log.KeyValue, 0, len(fields))
	for k, v := range fields {
		kvs = append(kvs, log.KeyValue{
			Key:   k,
			Value: convertValue(v),
		})
	}
```

File: bridges/otellogrus/hook.go
Comment (pellared) [inline bridges/otellogrus/hook.go:61]: We miss testing of default configuration and when `WithInstrumentationScope` is passed.

File: bridges/otellogrus/hook_test.go
Comment (pellared) [inline bridges/otellogrus/hook_test.go:215]: Asserting `[]log.KeyValue` with testify may stop working as `log.KeyValue` should be compared using `Equal`. Create a helper e.g.:

```go
func assertKeyValues(t *testing.T, want, got []log.KeyValue) {
	t.Helper()
	if !slices.EqualFunc(want, got, log.KeyValue.Equal) {
		t.Errorf("KeyValues are not equal:\nwant: %v\ngot:  %v", want, got)
	}
}
```

File: bridges/otellogrus/hook_test.go
Comment (pellared) [inline bridges/otellogrus/hook_test.go:115]: `log.Record` is not comparable. It would be safer to not use `testify` which uses reflection to check for eqaulity and create a helper function instead.

**Review by pellared — COMMENTED** [review]
It would be nice to increase the test coverage.

A test example and benchmarks could be added as separate PRs.

File: CODEOWNERS
Comment (pellared) [inline CODEOWNERS:25]:
```suggestion
bridges/otellogrus/                                                     @open-telemetry/go-approvers @dmathieu @pellared
```

I can also be a codeowner. I used to use `logrus`.

File: bridges/otellogrus/hook.go
Comment (pellared) [inline bridges/otellogrus/hook.go:246]: not tested

File: bridges/otellogrus/hook.go
Comment (pellared) [inline bridges/otellogrus/hook.go:234]: not tested

File: bridges/otellogrus/hook.go
Comment (pellared) [inline bridges/otellogrus/hook.go:92]: not tested

File: bridges/otellogrus/hook.go
Comment (dmathieu) [reply bridges/otellogrus/hook.go:246]: I extracted this condition from #5357, but it appears to be redundant for logrus, as we already check for `nil` type earlier.
I've removed it.

File: bridges/otellogrus/hook.go
Comment (dmathieu) [reply bridges/otellogrus/hook.go]: Opened #5428.

**Review by pellared — APPROVED** [review]

File: bridges/otellogrus/hook.go
Comment (MadVikingGod) [inline bridges/otellogrus/hook.go:36]: It looks like the transformation is more in-depth than this lets on.  I would suggest making this more generic so it doesn't get any further out of date.

File: bridges/otellogrus/hook_test.go
Comment (MadVikingGod) [inline bridges/otellogrus/hook_test.go:343]: Some suggested tests:
- []any with heterogeneous types  
- map[string]any with heterogeneous types  
- map[Struct]string

These are all odd formulations in otel, but are allowed by logrus.

**Review by MadVikingGod — APPROVED** [review]

File: bridges/otellogrus/go.mod
Comment (XSAM) [inline bridges/otellogrus/go.mod:19]:
```suggestion
	go.opentelemetry.io/otel v1.26.0 // indirect
	go.opentelemetry.io/otel/metric v1.26.0 // indirect
	go.opentelemetry.io/otel/trace v1.26.0 // indirect
```

Should we use the latest version of otel?

File: bridges/otellogrus/hook.go
Comment (dmathieu) [reply bridges/otellogrus/hook.go:36]: I based this after the comment in the otelslog bridge, but being more generic makes sense.

Comment (pellared) [reply bridges/otellogrus/hook.go]: Personally, I like the current description. I find it very concise and precise. If we change the transformation behavior I would expect to have the these docs updated and a mention in the changelog.

File: bridges/otellogrus/hook.go
Comment (pellared) [inline bridges/otellogrus/hook.go:28]:
```suggestion
// into a string attribute if there is no matching type.
```

File: bridges/otellogrus/hook.go
Comment (pellared) [inline bridges/otellogrus/hook.go:27]: This is also transforming the Body - not only the Attributes.

**Review by pellared — APPROVED** [review]

**Comment by pellared** [comment]
Blocked by:
- https://github.com/open-telemetry/opentelemetry-go/pull/5306

**Pull request merged**
